docs/examples/example_survival.ipynb

```diff
@@ -0,0 +1,206 @@
+    "for c in [\"race\", \"wexp\", \"mar\", \"paro\", \"censored\"]:\n",
+    "    X[c] = X[c].astype(\"category\")\n",
+    "\n",
+    "y = rossi[\"week\"]\n",
+    "w = rossi[\"fin\"]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Creating the base models\n",
+    "\n",
+    "Some survival models face two primary challenges:\n",
+    "\n",
+    "* **Time to event prediction:** Certain models, particularly those relying on the [Cox Proportional Hazards Model](https://en.wikipedia.org/wiki/Proportional_hazards_model), may not directly provide the time to event prediction. Instead, they yield hazard proportions; however, the desired output is the more useful and direct measure of time to the event.\n",
+    "\n",
+    "* **Censoring Support:** Survival analysis involves dealing with censored data, where the outcome variable ``y`` may not always be a singular real number for each observation and could be an interval instead.\n",
+    "\n",
+    "\n",
+    "To address these issues, we need to construct a model to act as the nuisance model. Here's how we can address the associated problems:\n",
+    "\n",
+    "* **Time to event prediction:** The principle to remember is that this nuisance model, when invoking the ``predict`` method, must return a time to event prediction for each observation. In this case, we will employ [``xgboost`` with Accelerated Failure Time](https://xgboost.readthedocs.io/en/stable/tutorials/aft_survival_analysis.html), since its output is inherently the time to event. However, other models may necessitate integration over the survival function or necessitate other transformations.\n",
+    "\n",
+    "* **Censoring Support:** For handling this issue, we keep ``y`` as is. If the event was observed, the value represents when it occurred, and if not observed, it is the censoring time. We then augment the ``X`` dataset with an extra column (which won't be passed to the model), labeled ``\"censored\"``, indicating whether the observation was censored or not. This column aids in constructing the upper and lower bounds for the output label from the model.\n",
+    "\n",
+    "\n",
+    "We suggest leveraging the base classes from ``sklearn`` to ensure that some essential methods needed for the MetaLearner are pre-defined and can be utilized effectively."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import xgboost as xgb\n",
+    "from sklearn.base import BaseEstimator, RegressorMixin, check_is_fitted\n",
+    "import numpy as np\n",
+    "\n",
+    "\n",
+    "class SurvivalModel(RegressorMixin, BaseEstimator):\n",
+    "    def __init__(self, n_estimators=100, booster_params=None) -> None:\n",
+    "        self.booster_params = booster_params\n",
+    "        self.n_estimators = n_estimators\n",
+    "\n",
+    "    def fit(self, X, y):\n",
+    "        common_params = {\n",
```

Comment:
I am just curious why don't you want to set these already at the initialization of the model class?

Reply:
This is because `sklearn` relies on having the `__init__` method with just saving the parameters with the same name as in the class signature, see [here](https://scikit-learn.org/stable/developers/develop.html#instantiation). This is needed because internally we rely on the `cross_validate` function and to clone the estimators `set_params` and similar need to be defined (this is done by inheriting from `BaseEstimator`).

Changing it to the `__init__` raises this error:
```python
RuntimeError: Cannot clone object SurvivalModel(booster_params={'aft_loss_distribution': 'normal',
                              'aft_loss_distribution_scale': 1.2,
                              'eval_metric': 'aft-nloglik',
                              'objective': 'survival:aft'},
              n_estimators=5), as the constructor either does not set or modifies parameter booster_params
```

Reply:
I see, makes sense, thanks! 👍